src/NUnitFramework/tests/Internal/EventListenerTextWriterTests.cs

```diff
@@ -0,0 +1,336 @@
+﻿// ***********************************************************************
+// Copyright (c) 2007 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using NUnit.Framework.Interfaces;
+
+namespace NUnit.Framework.Internal
+{
+    [TestFixture]
+    public class EventListenerTextWriterTests
+    {
+        private static readonly string NL = Environment.NewLine;
+
+        TestListenerHelper TestResult;
+        TextWriter TestWriter;
+
+        [SetUp]
+        public void SetUp()
+        {
+            TestResult = new TestListenerHelper();
+            TestExecutionContext.CurrentContext.Listener = TestResult;
```

Comment:
The scenario is that I'm trying to test the EventListenerTextWriter class - that is where the newline issue occurs.

The constructor for this class is EventListenerTextWriter(string streamName, TextWriter defaultWriter). It will then communicate with a listener by calling TestExecutionContext.CurrentContext.Listener.OnTestEvent().

So,  I cannot call the listener directly because it is hidden inside the class. How would you suggest I proceed?